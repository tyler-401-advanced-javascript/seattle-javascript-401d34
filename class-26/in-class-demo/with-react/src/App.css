body {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.center {
  display: flex;
  flex-direction: column;
}

.count {
  font-size: 24px;
}

#number {
  font-weight: bold;
}
