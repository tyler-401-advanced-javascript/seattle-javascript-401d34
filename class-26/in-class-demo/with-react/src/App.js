import React from 'react';
import logo from './logo.svg';
import './App.css';

class App extends React.Component {
  constructor () {
    super()
    this.state = {
      count: 0
    };
  }

  render () {
    return (
      <div className="App center">
        <h1>My Awesome Website!</h1>
        <h2>You won't believe how awesome it is</h2>
        <div class="count">
          Current count: <span id="number">{this.state.count}</span>
        <button id="increment" onClick={() => this.setState({
            count: this.state.count + 1
          })}>Increment</button>
        <button id="decrement" onClick={() => this.setState({
            count: this.state.count - 1
          })}>Decrement</button>
        </div>
      </div>
    );
  }
}

export default App;
