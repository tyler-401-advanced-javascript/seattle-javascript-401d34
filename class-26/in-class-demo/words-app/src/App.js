import React from 'react';
import './App.css';

import Header from './Header'
import Footer from './Footer'

class App extends React.Component {
  constructor () {
    super()
    const params = new URL(document.location).searchParams
    this.state = {
      words: params.get('words').split(',')
    }
    // this.handleChange = this.handleChange.bind(this)
    // this.handleNewWord = this.handleNewWord.bind(this)
  }

  traditionalClassMethod () {
    // you'll need to .bind(this) in the constructor so you get
    // "this" to refer to the right place
  }

  arrowFunctionClassMethod = () => {
    // you DON'T need to .bind(this) in the constructor, because arrow functions
    // have lexically scoped "this"
  }

  handleChange = event => {
    event.preventDefault();
    this.setState({
      newWord: event.target.value
    });
  }

  handleNewWord = event => {
    event.preventDefault();
    this.setState({
      words: [...this.state.words, this.state.newWord]
    })
  }

  render () {
    return (
      <>
        <Header />
        <div className="App">
          <form onSubmit={this.handleNewWord}>
            <input onChange={this.handleChange} placeholder="enter a new word here" />
            <input type="submit" value="Enter new word" />
          </form>
          <ul>
            {this.state.words.map(word => {
              return <li>{word}</li>
            })}
          </ul>
        </div>
        <Footer />
      </>
    );
  }
}

export default App;
