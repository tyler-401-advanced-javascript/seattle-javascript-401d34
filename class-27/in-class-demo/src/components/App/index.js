import React from 'react';
import './App.css';
import Thing from '../Thing'

function App() {
  return (
    <div className="App">
      <h1>My App</h1>
      <Thing />
    </div>
  );
}

export default App;
