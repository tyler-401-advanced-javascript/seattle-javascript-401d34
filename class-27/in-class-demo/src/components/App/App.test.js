import React from 'react'
import App from './'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('<App />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('is alive at application start', () => {
    const component = shallow(<App />)
    expect(component.find('h1')).toExist()
  })
})
