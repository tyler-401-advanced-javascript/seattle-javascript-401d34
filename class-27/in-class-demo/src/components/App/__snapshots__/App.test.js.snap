// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <h1>
    My App
  </h1>
  <section
    className="Thing"
  >
    <p>
      My wonderful component
    </p>
    <span>
      Stuff: 
      true
    </span>
    <button
      id="myButton"
      onClick={[Function]}
    >
      Clicky
    </button>
  </section>
</div>
`;
